FrameworksAndDrivers/Web/DotLanches.Api/Controllers/PagamentoController.cs

```diff
@@ -25,19 +25,35 @@ public PagamentoController(IPagamentoRepository pagamentoRepository, IPedidoRepo
         }
 
         /// <summary>
-        /// Envia os dados para o meio de pagamento e retorna a senha para retirar o pedido.
+        /// Envia os dados para o meio de pagamento e retorna o QR Code para o cliente.
         /// </summary>
         /// <param name="pagamentoRequest">Dados de requisição do pagamento</param>
+        /// <returns>QR Code para efetuar pagamento.</returns>
+        [HttpPost("QrCode")]
+        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> ExecutePayment([Required][FromBody] PagamentoRequestDto pagamentoRequest)
+        {
+            var adapterPagamento = new AdapterPagamentoController(_pagamentoRepository, _pedidoRepository, _checkout);
+            var qrCode = await adapterPagamento.ProcessPagamento(pagamentoRequest.IdPedido);
+            return Ok(qrCode);
+        }
+
+        /// <summary>
+        /// Recebe o status do pagamento e segue com a confirmação ou cancelamento do pedido.
+        /// </summary>
+        /// <param name="pagamentoResponse">Dados de requisição do pagamento</param>
         /// <returns>Situação do pagamento e senha para retirada do pedido.</returns>
-        [HttpPost]
+        [HttpPost("Confirmar")]
         [ProducesResponseType(typeof(PagamentoViewModel), StatusCodes.Status200OK)]
         [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
         [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
-        public async Task<IActionResult> ExecutePayment([Required][FromBody] PagamentoRequestDto pagamentoRequest)
+        public async Task<IActionResult> PaymentResponseHandler([Required][FromBody] PagamentoResponseDto pagamentoResponse)
```

Comment:
O nome ResponseHandler não fica muito claro o que o endpoint faz, e também temos a convenção de utilizar o nome das nossas entidades em PT.

Acredito que como fazemos uma decisão baseado no sucesso da operação, esse endpoint poderia ser o de processar pagamento:
```suggestion
        public async Task<IActionResult> ProcessPagamento([Required][FromBody] PagamentoResponseDto pagamentoResponse)
```